nestkernel/source_table.cpp

```diff
@@ -243,18 +243,22 @@ nest::SourceTable::compute_buffer_pos_for_unique_secondary_sources( const thread
       const thread source_rank = kernel().mpi_manager.get_process_id_of_node_id( cit->first );
       const size_t event_size = kernel().model_manager.get_secondary_event_prototype( cit->second, tid ).size();
 
-      buffer_pos_of_source_node_id_syn_id.insert( std::make_pair(
-        pack_source_node_id_and_syn_id( cit->first, cit->second ), recv_buffer_position_by_rank[ source_rank ] ) );
-      recv_buffer_position_by_rank[ source_rank ] += event_size;
+      buffer_pos_of_source_node_id_syn_id.insert(
+        std::make_pair( pack_source_node_id_and_syn_id( cit->first, cit->second ),
+          recv_counts_secondary_events_in_int_per_rank[ source_rank ] ) );
+
+      recv_counts_secondary_events_in_int_per_rank[ source_rank ] += event_size;
     }
 
-    // compute maximal chunksize per source rank and determine global
-    // maximal chunksize; will need to be taken into account when
-    // filling secondary_recv_buffer_pos_ in ConnectionManager
-    std::vector< long > max_uint_count(
-      1, *std::max_element( recv_buffer_position_by_rank.begin(), recv_buffer_position_by_rank.end() ) );
-    kernel().mpi_manager.communicate_Allreduce_max_in_place( max_uint_count );
-    kernel().mpi_manager.set_chunk_size_secondary_events_in_int( max_uint_count[ 0 ] + 1 );
+    // each chunk needs to contain one additional int that can be used
+    // to communicate whether waveform relaxation has converged
+    for ( thread rank = 0; rank < kernel().mpi_manager.get_num_processes(); ++rank )
+    {
+      recv_counts_secondary_events_in_int_per_rank[ rank ] += 1;
+    }
```

Comment:
I think a range-based loop is more readable here.
```suggestion
    for ( auto& recv_count : recv_counts_secondary_events_in_int_per_rank )
    {
      ++recv_count;
    }
```